### PR TITLE
fix(auth): refactor auth

### PR DESCRIPTION
**Pull request by gund**
- [x] Unify auth package to use Oauth
- [x] Add support for Authorization Code and Password Grant flows
- [x] Setup auth features for SAPI and BAPI in `application` package
- [x] Fix build/tests/ci

Labs PR: https://github.com/spryker/oryx-labs/pull/8

### CODE REVIEW

File: libs/platform/auth/package.json [hunk 1]
```diff
@@ -1,33 +1,40 @@
 {
   "name": "@spryker-oryx/auth",
-  "version": "0.1.0-rc.0",
+  "version": "1.0.0-alpha.0",
```
Comment (tolerants) [inline libs/platform/auth/package.json]: Can you please check? :)

File: libs/platform/auth/project.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
+  "sourceRoot": "libs/platform/router/src",
```
Comment (tolerants) [inline libs/platform/auth/project.json]:
```suggestion
  "sourceRoot": "libs/platform/auth/src",
```

File: libs/platform/auth/project.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      "executor": "./tools/executors/components-library:build"
+    },
+    "paths": {
+      "executor": "./tools/executors/components-library:paths"
+    },
+    "lint": {
+      "executor": "@nrwl/linter:eslint",
+      "outputs": ["{options.outputFile}"],
+      "options": {
+        "lintFilePatterns": ["libs/platform/router/**/*.ts"]
```
Comment (tolerants) [inline libs/platform/auth/project.json]:
```suggestion
        "lintFilePatterns": ["libs/platform/auth/**/*.ts"]
```

File: libs/platform/auth/src/components/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+  @property() heading?: DirectiveResult | string;
+  @property() isLoading?: boolean;
+  @property() hasError?: boolean;
+  @property() emailName = 'email';
+  @property() passwordName = 'password';
+  @property() rememberMeName = 'rememberme';
+
+  @state() protected isDisabled = false;
+
+  #internals =
```
Comment (tolerants) [inline libs/platform/auth/src/components/login/login.component.ts]: But is it working in Safari?

Comment (gund) [reply libs/platform/auth/src/components/login/login.component.ts]: yes
image

File: libs/platform/auth/package.json (hunk 1 above)
Comment (gund) [reply libs/platform/auth/package.json]: Fixed to 0.1.0 as in all other packages.

File: libs/platform/auth/project.json (hunk 2 above)
Comment (gund) [reply libs/platform/auth/project.json]: Fixed

File: libs/platform/auth/project.json (hunk 3 above)
Comment (gund) [reply libs/platform/auth/project.json]: Fixed

File: libs/platform/auth/src/components/login/login.component.ts (hunk 4 above)
Comment (gund) [reply libs/platform/auth/src/components/login/login.component.ts]: Actually I had to drop it because storybook does not support this syntax in decorated classes 🤷

File: libs/platform/auth/package.json [hunk 5]
```diff
@@ -2,32 +2,39 @@
   "name": "@spryker-oryx/auth",
   "version": "0.1.0",
   "type": "module",
-  "description": "Authentication features for Oryx applications",
-  "author": "Spryker Systems GmbH",
-  "license": "See license in LICENSE.md",
+  "description": "Authentication functionality for Oryx framework",
+  "author": "Spryker",
```
Comment (tolerants) [inline libs/platform/auth/package.json]: Can you please revert this change?

File: libs/platform/auth/package.json [hunk 6]
```diff
@@ -2,32 +2,39 @@
...
+  "license": "MIT",
```
Comment (tolerants) [inline libs/platform/auth/package.json]: Can you please revert this change?

File: libs/platform/auth/package.json (hunk 5 above)
Comment (gund) [reply libs/platform/auth/package.json]: Fixed

File: libs/platform/auth/package.json (hunk 6 above)
Comment (gund) [reply libs/platform/auth/package.json]: Fixed

File: libs/platform/auth/src/components/logout/logout.def.ts [hunk 7]
```diff
@@ -9,7 +9,7 @@ declare global {
 
 export const authLogoutComponent = componentDef({
   name: 'oryx-auth-logout',
-  impl: () => import('./logout.component').then((m) => m.AuthLogoutComponent),
+  impl: () => import('./logout.component').then((m) => m.default),
```
Comment (dunqan) [inline libs/platform/auth/src/components/logout/logout.def.ts]: Why `default`?

**Review by dunqan — DISMISSED** [review]

**Review by tolerants — DISMISSED** [review]

File: libs/platform/auth/src/components/logout/logout.def.ts (hunk 7 above)
Comment (gund) [reply libs/platform/auth/src/components/logout/logout.def.ts]: Actually I would like to propose to use default for our dynamic imports to reduce amount of code needed to be written, and to add default extraction to the loader itself, so we can simply keep `import('./logout.component')` and no `.then()` chain is required.
So this was just a preparation for that stage =)

**Review by dunqan — DISMISSED** [review]

File: libs/platform/auth/src/components/logout/logout.def.ts (hunk 7 above)
Comment (dunqan) [reply libs/platform/auth/src/components/logout/logout.def.ts]: yeah, we should discuss it first then, instead of sneaking it into code base :P

Comment (tobi-or-not-tobi) [reply libs/platform/auth/src/components/logout/logout.def.ts]: Not a big deal to merge this PR now, but I agree we need to discuss this one. 

I wonder if there are any consequences. For example, in some cases we might have multiple exports in the component fil.

Comment (tobi-or-not-tobi) [reply libs/platform/auth/src/components/logout/logout.def.ts]: Anyway, let's rol it back as it's not related to auth topic

Comment (tobi-or-not-tobi) [reply libs/platform/auth/src/components/logout/logout.def.ts]: (i'm doing it so we can resolve this one and merge)

**Review by dunqan — APPROVED** [review]
